grails-app/controllers/de/iteratec/osm/result/ResultSelectionController.groovy

```diff
@@ -170,23 +170,41 @@ class ResultSelectionController extends ExceptionHandlerController {
             sendError(command)
             return
         }
-
         def dtos = performanceLoggingService.logExecutionTime(DEBUG, "getUserTimings for ${command as JSON}", 0, {
-            def userTimings = query(command, ResultSelectionType.UserTimings, { existing ->
-                projections {
-                    userTimings {
-                        groupProperty('name')
-                        groupProperty('type')
-                    }
-                }
-            })
+            def userTimings = getAllUserAndHeroTimings(command).findAll { it[1] != UserTimingType.HERO_MARK }
```

Comment:
Instead of using `findAll`, you could pass the desired UserTimingType(s) as an argument to the function and directly filter for them in the query.